internal/state/context.go

```diff
@@ -0,0 +1,26 @@
+package state
+
+import "context"
+
+type stateContextType struct{}
+
+var stateContextKey stateContextType
+
+func NewStateContext(ctx context.Context, s *State) context.Context {
```

Comment:
Would appreciate a docstring describing what the state context represents and how we use it internally

Reply:
Done